advanced_alchemy/base.py

```diff
@@ -291,167 +295,226 @@ def create_registry(
         type_annotation_map[Struct] = JsonB
     if custom_annotation_map is not None:
         type_annotation_map.update(custom_annotation_map)
-    return registry(metadata=meta, type_annotation_map=type_annotation_map)
+    return SQLAlchemyRegistry(metadata=meta, type_annotation_map=type_annotation_map)
 
 
 orm_registry = create_registry()
 
 
-class UUIDBase(_UUIDPrimaryKey, CommonTableAttributes, DeclarativeBase, AsyncAttrs):
+class MetadataRegistry:
+    """A registry for metadata."""
+
+    _instance: MetadataRegistry | None = None
+    _registry: dict[str | None, MetaData] = {None: orm_registry.metadata}
+
+    def __new__(cls) -> Self:
+        if cls._instance is None:
+            cls._instance = super().__new__(cls)
+            cls._registry = {}
+        return cast(Self, cls._instance)
+
+    def get(self, bind_key: str | None = None) -> MetaData:
+        """Get the metadata for the given bind key."""
+        return self._registry.setdefault(bind_key, MetaData(naming_convention=convention))
+
+    def set(self, bind_key: str | None, metadata: MetaData) -> None:
+        """Set the metadata for the given bind key."""
+        self._registry[bind_key] = metadata
+
+    def __iter__(self) -> Iterator[str | None]:
+        return iter(self._registry)
+
+    def __getitem__(self, bind_key: str | None) -> MetaData:
+        return self._registry[bind_key]
+
+    def __setitem__(self, bind_key: str | None, metadata: MetaData) -> None:
+        self._registry[bind_key] = metadata
+
+    def __contains__(self, bind_key: str | None) -> bool:
+        return bind_key in self._registry
+
+
+metadata_registry = MetadataRegistry()
+
+
+class AdvancedDeclarativeBase(DeclarativeBase):
+    """A subclass of declarative base that allows for overriding of the registry.
+
+    .. seealso::
+        :class:`sqlalchemy.orm.DeclarativeBase`
+    """
+
+    registry = orm_registry
+    __abstract__ = True
+    __metadata_registry__: MetadataRegistry = MetadataRegistry()
+    __bind_key__: str | None = None
+
+    def __init_subclass__(cls, **kwargs: Any) -> None:
+        bind_key = getattr(cls, "__bind_key__", None)
+        if bind_key is not None:
+            if bind_key not in cls.__metadata_registry__:
+                cls.__metadata_registry__[bind_key] = MetaData(naming_convention=convention)
+            cls.metadata = cls.__metadata_registry__[bind_key]
```

Comment:
```suggestion
        if bind_key is not None:
            cls.metadata = cls.__metadata_registry__.get(bind_key)
```

Correct me if I am wrong, if we are going with the current implementation for `MetadataRegistry.get`, that already does this for you.